file-formats/uipg/type/zif_aff_uipg_v1.intf.abap

```diff
@@ -0,0 +1,201 @@
+INTERFACE zif_aff_uipg_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    "! General information
+    BEGIN OF ty_general_information,
+
+      "! <p class="shorttext">Title</p>
+      "! Title visible to the enduser
+      "! $required
+      title       TYPE zif_aff_types_v1=>ty_description_100,
+
+      "! <p class="shorttext">Description</p>
+      "! Description for design-time tools
+      "! $required
+      description TYPE zif_aff_types_v1=>ty_description_100,
+
+      "! <p class="shorttext">Merge ID</p>
+      "! Merge ID for merging during run-time
+      merge_id    TYPE c LENGTH 35,
+
+    END OF ty_general_information.
+
+
+  "! <p class="shorttext">Catalog Type</p>
+  "! Catalog type
+  TYPES ty_catalog_type        TYPE c LENGTH 3.
+
+
+  "! <p class="shorttext">Tile Display Format</p>
+  "! Tile display format
+  "! $values {@link zif_aff_uipg_v1.data:co_tile_display_format}
+  "! $default {@link zif_aff_uipg_v1.data:co_tile_display_format.standard}
+  TYPES ty_tile_display_format TYPE c LENGTH 2.
+
+
+  CONSTANTS:
+    "! <p class="shorttext">Tile Display Format</p>
+    "! Tile display format
+    BEGIN OF co_tile_display_format,
+
+      "! <p class="shorttext">Standard Tile</p>
+      "! Standard tile
+      standard  TYPE ty_tile_display_format VALUE 'TI',
+
+      "! <p class="shorttext">Wide Tile</p>
+      "! Wide tile
+      wide      TYPE ty_tile_display_format VALUE 'TW',
+
+      "! <p class="shorttext">Flat Tile</p>
+      "! Flat tile
+      flat      TYPE ty_tile_display_format VALUE 'FT',
+
+      "! <p class="shorttext">Flat Wide Tile</p>
+      "! Flat wide tile
+      flat_wide TYPE ty_tile_display_format VALUE 'FW',
+
+      "! <p class="shorttext">Link</p>
+      "! Link
+      link      TYPE ty_tile_display_format VALUE 'LK',
+
+    END OF co_tile_display_format.
+
+
+  TYPES:
+    "! <p class="shorttext">Catalog Item Reference</p>
+    "! Catalog item reference
+    BEGIN OF ty_catalog_item_reference,
+
+      "! <p class="shorttext">Catalog ID</p>
+      "! Catalog ID
+      "! $required
+      catalog_id   TYPE c LENGTH 35,
+
+      "! <p class="shorttext">Catalog Type</p>
+      "! Catalog type
+      "! $default 'CAT'
```

Comment:
I am not sure whether I provided misleading feedback. Most probably I would have set "standard" as default.

Reply:
We currently want the value to directly reflect the value in the database.

The reasonsing is that there are in theory additional values possible except for standard, but I don't want to maintain a mapping for all of them and I don't want the schema to be invalied, when the value is not maintained.

99% of all objects should use this CAT value here and with this it is possible to ignore this value for the 99% case and just to maintain it for the rare exception case.